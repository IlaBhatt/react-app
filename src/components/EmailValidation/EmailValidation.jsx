// author @POD-1
import React, { useEffect } from "react";
import {
  Card,
  Typography,
  CardContent,
  CardActions,
  Button,
} from "@material-ui/core";
import { useNavigate } from "react-router-dom";
import {
  getUserFromEmail,
  getToken,
  sendEmailVerification,
} from "../../UserService";

const EmailValidation = () => {
  const navigate = useNavigate();
  getToken();
  // get current user email
  const email = localStorage.getItem("email");
  if (localStorage.getItem("email") === undefined) {
    console.log("email not saved in registration");
    return;
  }

  // calling once to save
  useEffect(() => {
    getUserFromEmail(email).then((response) =>
      response.json().then(async (data) => {
        // get id of user
        await sendEmailVerification(data[0].id)
          .then((response) => {
            if (response.status === 204) {
              // email verification sent
              console.log("Verification Email Sent");
            } else {
              // error sending email verification
              console.log("Verification Email Not sent");
            }
          })
          .catch((error) => console.log(error));
      }, [])
    );
  }, []);

  const handleProceed = () => {
    //! onProceed
    // get user and check if email verified or not
    getToken();
    getUserFromEmail(email).then((response) =>
      response.json().then((data) => {
        if (data[0].emailVerified === false) {
          console.log("email not verified");
          // toast("email not verified")
        } else {
          console.log("email verified");
          navigate("/");
        }
      })
    );
  };
  return (
    <>
      <div
        styles={{
          height: "100%",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          fontSize: "30",
          backgroundColor: "#fdfdfd",
        }}
      >
        {/* height: 280,
            paddingTop: 30,
            paddingLeft: 50,
            paddingRight: 50,
            marginTop: 100,
            backgroundColor: "white", */}
        <Card
          variant="outlined"
          style={{
            width: "30%",
            height: "50%",
            padding: "1% 4%",
            margin: "auto",
            marginTop: "5%",
            backgroundColor: "white",
          }}
        >
          <CardContent style={{ padding: 20 }}>
            <Typography gutterBottom variant="h5" component="h2" align="center">
              Hi {localStorage.getItem("firstName")},
            </Typography>
            <Typography
              variant="body2"
              color="textSecondary"
              component="p"
              align="center"
              paddingTop="30"
            >
              An email with confirmation link has been sent to{" "}
              <a href="">{localStorage.getItem("email")}</a>. Please verify by
              your identity by clicking on the link and come back here.
            </Typography>
          </CardContent>

          <CardActions>
            <Button
              variant="contained"
              onClick={handleProceed}
              style={{
                display: "flex",
                justifyContent: "center",
                alignItems: "center",
                color: "#FFFFFF",
                backgroundColor: "#24a0ed",
                paddingLeft: "35%",
                paddingRight: "35%",
              }}
            >
              Verify Email
            </Button>
          </CardActions>
        </Card>
      </div>
    </>
  );
};

export default EmailValidation;
