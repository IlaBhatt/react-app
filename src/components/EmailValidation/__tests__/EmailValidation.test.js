// author @POD-1
import { shallow, render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import EmailValidation from "../EmailValidation";
import { Button } from "@material-ui/core";
import { Card, Typography, CardContent, CardActions } from "@mui/material";
import { BrowserRouter as Router } from "react-router-dom";

describe("EmailValidation", function () {
  let emailvalidation;
  beforeEach(() => {
    emailvalidation = shallow(
      <Router>
        <EmailValidation />
      </Router>
    );
  });

  it("renders email verification correctly", () => {
    const tree = renderer.create(emailvalidation).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the card correctly", () => {
    const component = shallow(<Card />);
    expect(component.length).toBe(1);
  });
  it("renders the card content correctly", () => {
    const component = shallow(<CardContent />);
    expect(component.length).toBe(1);
  });

  it("renders the typography tag correctly", () => {
    const component = shallow(<Typography />);
    expect(component.length).toBe(1);
  });

  it("renders the card actions tag correctly", () => {
    const component = shallow(<CardActions />);
    expect(component.length).toBe(1);
  });

  describe("Button", function () {
    let mountedButton;

    it("renders without crashing", () => {
      mountedButton = render(<Button />);
      expect(mountedButton.length).toBe(1);
    });

    it("matches snapshot", () => {
      const tree = renderer.create(<Button id="proceed"></Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders the proceed button", () => {
      const btn = render(<Button id="proceed" />);
      expect(btn.length).toBe(1);
    });
  });
});
