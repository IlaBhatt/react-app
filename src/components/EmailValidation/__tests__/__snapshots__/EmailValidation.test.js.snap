// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailValidation Button matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  id="proceed"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`EmailValidation renders email verification correctly 1`] = `
<div
  styles={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fdfdfd",
      "display": "flex",
      "fontSize": "30",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "white",
        "height": "50%",
        "margin": "auto",
        "marginTop": "5%",
        "padding": "1% 4%",
        "width": "30%",
      }
    }
  >
    <div
      className="MuiCardContent-root"
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Hi 
        ,
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        paddingTop="30"
      >
        An email with confirmation link has been sent to
         
        <a
          href=""
        />
        . Please verify by your identity by clicking on the link and come back here.
      </p>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#24a0ed",
            "color": "#FFFFFF",
            "display": "flex",
            "justifyContent": "center",
            "paddingLeft": "35%",
            "paddingRight": "35%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Verify Email
        </span>
      </button>
    </div>
  </div>
</div>
`;
